### PR TITLE
fix to be able to search for numeric fields with Ickle full text range query

### PR DESCRIPTION
**Pull request by kazuhira-r**
In current Ickle, full text range query like [this blog entry](http://blog.infinispan.org/2016/12/meet-ickle.html) does not work.
```
from com.acme.Transaction where amount : [40 to 90}
```
I think that the cause is as follows.
- QueryRenderer class can not get value of condition of range query
- [QueryRendererDelegateImpl class expects the search condition to be a string](https://github.com/infinispan/infinispan/blob/9.0.0.Final/object-filter/src/main/java/org/infinispan/objectfilter/impl/syntax/parser/QueryRendererDelegateImpl.java#L387-L388)
- [QueryRendererDelegateImpl class expects an analyzed field, but the numeric field is not analyzed](https://github.com/infinispan/infinispan/blob/9.0.0.Final/object-filter/src/main/java/org/infinispan/objectfilter/impl/syntax/parser/QueryRendererDelegateImpl.java#L389)

It behaves differently for other Query APIs.

For example, the following code created based on [QueryStringTest](https://github.com/infinispan/infinispan/blob/9.0.0.Final/query/src/test/java/org/infinispan/query/dsl/embedded/QueryStringTest.java) will work
```java
   public void testFullTextRangeAsDsl() throws Exception {
      QueryFactory qf = getQueryFactory();
      Query q =
              qf.from(getModelFactory().getTransactionImplClass())
                      .having("amount")
                      .between(23, 45)
                      .build();
      List<Transaction> list = q.list();
      assertEquals(2, list.size());
   }

   public void testFullTextRangeHs() throws Exception {
      SearchManager searchManager = Search.getSearchManager((Cache) getCacheForQuery());
      org.apache.lucene.search.Query luceneQuery =
              searchManager
                      .buildQueryBuilderForClass(getModelFactory().getTransactionImplClass())
                      .get()
                      .range()
                      .onField("amount")
                      .from(23.0).to(45.0)
                      .createQuery();
      CacheQuery q = searchManager.getQuery(luceneQuery, getModelFactory().getTransactionImplClass());

      List<Transaction> list = q.list();
      assertEquals(2, list.size());
   }
```

The following code does not work.(A NullPointerException is thrown.)
```java
   public void testFullTextRangeIckle() throws Exception {
      QueryFactory qf = getQueryFactory();

      Query q = qf.create("from " + getModelFactory().getTransactionTypeName() + " t where t.amount : [23 to 45]");

      List<Transaction> list = q.list();
      assertEquals(2, list.size());
   }
```

I think like this, but how is it?

### CODE REVIEW

**Comment by anistor** [comment]
Looking...

File: object-filter/src/main/antlr3/org/infinispan/objectfilter/impl/ql/parse/QueryRenderer.g [hunk 1]
```diff
@@ -361,8 +361,8 @@ ftClause
          { delegate.predicateFullTextRegexp($REGEXP_LITERAL.text); }
    |  ^(FT_RANGE startRange=(LSQUARE | LCURLY) lower=ftRangeBound upper=ftRangeBound endRange=(RSQUARE | RCURLY))
          { delegate.predicateFullTextRange($startRange.type == LSQUARE,
-                                     $lower.tree.getType() == ASTERISK ? null : $lower.text,
```
Comment (anistor) [inline object-filter/src/main/antlr3/org/infinispan/objectfilter/impl/ql/parse/QueryRenderer.g]: I could not figure out what is the purpose of this change.

**Comment by anistor** [comment]
RemoteNonIndexedQueryStringTest.testFullTextRangeWildcard is failing:
```org.infinispan.client.hotrod.exceptions.HotRodClientException:Request for messageId=67 returned server error (status=0x85): java.lang.IllegalStateException: The cache must be indexed in order to use full-text queries.```

File: object-filter/src/main/antlr3/org/infinispan/objectfilter/impl/ql/parse/QueryRenderer.g (hunk 1 above)
Comment (kazuhira-r) [inline object-filter/src/main/antlr3/org/infinispan/objectfilter/impl/ql/parse/QueryRenderer.g]: If I return this change, in the code below
```java
   public void testFullTextRangeIckle() throws Exception {
      QueryFactory qf = getQueryFactory();

      Query q = qf.create("from " + getModelFactory().getTransactionTypeName() + " t where t.amount : [23 to 45]");

      List<Transaction> list = q.list();
      assertEquals(2, list.size());
   }
```

A NullPointerException is thrown.
```
java.lang.NullPointerException
	at org.infinispan.objectfilter.impl.ql.parse.QueryRenderer.ftClause(QueryRenderer.java:8657)
	at org.infinispan.objectfilter.impl.ql.parse.QueryRenderer.fullTextExpression(QueryRenderer.java:8329)
	at org.infinispan.objectfilter.impl.ql.parse.QueryRenderer.predicate(QueryRenderer.java:3285)
	at org.infinispan.objectfilter.impl.ql.parse.QueryRenderer.searchCondition(QueryRenderer.java:3007)
	at org.infinispan.objectfilter.impl.ql.parse.QueryRenderer.whereClause(QueryRenderer.java:657)
	at org.infinispan.objectfilter.impl.ql.parse.QueryRenderer.querySpec(QueryRenderer.java:510)
	at org.infinispan.objectfilter.impl.ql.parse.QueryRenderer.queryStatement(QueryRenderer.java:379)
	at org.infinispan.objectfilter.impl.ql.parse.QueryRenderer.queryStatementSet(QueryRenderer.java:292)
	at org.infinispan.objectfilter.impl.ql.parse.QueryRenderer.statement(QueryRenderer.java:220)
	at org.infinispan.objectfilter.impl.ql.QueryParser.render(QueryParser.java:88)
	at org.infinispan.objectfilter.impl.ql.QueryParser.parseQuery(QueryParser.java:70)
	at org.infinispan.objectfilter.impl.syntax.parser.IckleParser.parse(IckleParser.java:19)
	at org.infinispan.query.dsl.embedded.impl.QueryEngine.lambda$parse$1(QueryEngine.java:663)
	at org.infinispan.query.dsl.embedded.impl.QueryCache.lambda$get$0(QueryCache.java:79)
	at org.infinispan.cache.impl.TypeConverterDelegatingAdvancedCache.lambda$convertFunction$1(TypeConverterDelegatingAdvancedCache.java:101)
	at java.util.concurrent.ConcurrentMap.computeIfAbsent(ConcurrentMap.java:324)
	at org.infinispan.cache.impl.AbstractDelegatingCache.computeIfAbsent(AbstractDelegatingCache.java:343)
	at org.infinispan.cache.impl.TypeConverterDelegatingAdvancedCache.computeIfAbsent(TypeConverterDelegatingAdvancedCache.java:161)
	at org.infinispan.query.dsl.embedded.impl.QueryCache.get(QueryCache.java:79)
	at org.infinispan.query.dsl.embedded.impl.QueryEngine.parse(QueryEngine.java:663)
	at org.infinispan.query.dsl.embedded.impl.DelegatingQuery.<init>(DelegatingQuery.java:38)
	at org.infinispan.query.dsl.embedded.impl.EmbeddedQueryFactory.create(EmbeddedQueryFactory.java:24)
	at org.infinispan.query.dsl.embedded.impl.EmbeddedQueryFactory.create(EmbeddedQueryFactory.java:11)
```

**Comment by kazuhira-r** [comment]
I'm sorry.
I added a test case to RemoteNonIndexedQueryStringTest.testFullTextRangeWildcard.

File: object-filter/src/main/antlr3/org/infinispan/objectfilter/impl/ql/parse/QueryRenderer.g (hunk 1 above)
Comment (anistor) [inline object-filter/src/main/antlr3/org/infinispan/objectfilter/impl/ql/parse/QueryRenderer.g]: got it! thanks!

File: object-filter/src/main/java/org/infinispan/objectfilter/impl/logging/Log.java [hunk 2]
```diff
@@ -96,4 +96,11 @@
 
    @Message(id = 28526, value = "Invalid query: %s; Parser error messages: %s.")
    ParsingException getQuerySyntaxException(String query, List<?> parserErrorMessages);
+
+   @Message(id = 28527, value = "No relational queries can be applied to property '%2$s' in type %1$s since the property is indexed.")
```
Comment (anistor) [inline object-filter/src/main/java/org/infinispan/objectfilter/impl/logging/Log.java]: This message does not make much sense. In fact, this error can never be thrown.

File: object-filter/src/main/java/org/infinispan/objectfilter/impl/syntax/parser/QueryRendererDelegateImpl.java [hunk 3]
```diff
@@ -407,6 +407,18 @@ private void checkAnalyzed(PropertyPath<?> propertyPath, boolean expectAnalyzed)
       }
    }
 
+   private void checkIndexed(PropertyPath<?> propertyPath, boolean expectIndexed) {
```
Comment (anistor) [inline object-filter/src/main/java/org/infinispan/objectfilter/impl/syntax/parser/QueryRendererDelegateImpl.java]: The ```expectedIndexed``` arg should always be true.

**Comment by anistor** [comment]
I've fixed the review comments myself, squashed the commits and integrated. Thanks @kazuhira-r !

**Comment by anistor** [comment]
I've also reported this in jira just in case somebody else stumbles upon the same problem: https://issues.jboss.org/browse/ISPN-7718

**Comment by kazuhira-r** [comment]
Thank you for your review and integration!
